transforms/dialog-full-screen-app-wrapper/dialog-full-screen-app-wrapper.js

```diff
@@ -0,0 +1,184 @@
+      const { declarations } = vd.value;
+
+      if (declarations.length && declarations[0].id.name === expressionContainerName) {
+        filteredDeclarations.push(declarations[0].init);
+      }
+    });
+
+    return filteredDeclarations;
+  }
+
+  // compute whether the given declaration already composes AppWrapper
+  const computeShouldUpdate = (declarations, appWrapperIdentifier) => {
+    let result = false;
+    declarations.forEach((declaration) => {
+      // declaration is jsx element
+      if (declaration.openingElement) {
+        result = declaration.openingElement.name.name !== appWrapperIdentifier;
+      } else if (declaration.body.openingElement) {
+        // declaration is function with no return
+        result = declaration.body.openingElement.name.name !== appWrapperIdentifier;
+      } else {
+        // declaration is function with return
+        result = declaration.body.body[0].argument.openingElement.name.name !== appWrapperIdentifier;
+      }
+    })
+
+    return result;
+  }
+
+  // adds the AppWrapper as a child of DialogFullScreen
+  const addAppWrapper = (dialogFullScreen, addedImport) => {
+    let appWrapperIdentifier = "AppWrapper";
+
+    // if there already was an import
+    if (!addedImport) {
+      const appWrapperImport = root.find(j.ImportDeclaration, {
+        source: {
+          type: "Literal",
+          value: "carbon-react/lib/components/app-wrapper",
+        },
+      });
+
+      // find identifier used for import
+      appWrapperIdentifier = appWrapperImport.get('specifiers', 0, 'local', 'name').value;
+    }
+    
+    const dialogFullScreenAppWrapper = j(dialogFullScreen).find(
+      j.JSXOpeningElement,
+      {
+        name: {
+          type: "JSXIdentifier",
+          name: appWrapperIdentifier,
+        },
+      }
+    );
+
+    // if app wrapper already exists within dialog, do nothing
+    if (dialogFullScreenAppWrapper.size()) {
+      return;
+    }
+    
+    const dialogFullScreenJSXExpressionContainer = j(dialogFullScreen).find(
+      j.JSXExpressionContainer
+    );
+      
+    let shouldUpdate = false;
+
+    // if there are no expression containers or multiple instances then add wrapper
+    if (dialogFullScreenJSXExpressionContainer.size() !== 1) {
+      shouldUpdate = true;
+    } else {
+    // else there is a single expression container
+      const expressionContainerName = dialogFullScreenJSXExpressionContainer.get('expression', 'name').value;
+      
+      const variableDeclarations = filteredDeclarations(root.find(j.VariableDeclaration), expressionContainerName);
+
+      shouldUpdate = !variableDeclarations.length || computeShouldUpdate(variableDeclarations, appWrapperIdentifier);
+    }
+
+    if (shouldUpdate) {
+      // AppWrapper tags
+      const appWrapperOpeningTag = j.jsxOpeningElement(j.jsxIdentifier(appWrapperIdentifier));
+      const appWrapperClosingTag = j.jsxClosingElement(j.jsxIdentifier(appWrapperIdentifier));
+    
```

Comment:
I think somehow here you need to work out if `DialogFullScreen` opening element is on a line by itself, and if so add the `AppWrapper` on the next line, and vice versa for the closing element.

That should sort my comment above on the test cases

Reply:
I could do I just feel it's adding extra complexity to something that is already catered for by lint/prettier

Reply:
As above, what if they don't have lint/prettier on their project?

Reply:
They will still have to go through and fix indentations then if neither are running on their project. I don't mind adding it just feel it's not needed as if they have a linter it will complain and fix, if not then any code transformed won't throw an error

Reply:
Fair play, if we didn't have the time constraints at the moment I'd probably say it's worth it but given how long you've already spent on this lets just leave it then and move on

Reply:
Yeah I'll be happy to add it in at a later date if it becomes necessary, might even be worth adding to the shared methods